### ISSUE TITLE
feat(metrics): Add more metrics (e.g. duration)

### ISSUE THREAD

**Issue by wei840222** [TwiN/gatus#274]
### Describe the feature request

In this project https://github.com/prometheus/blackbox_exporter, many metrics like probe_success, probe_duration etc... provided. And in dns probe https://github.com/prometheus/blackbox_exporter/blob/master/prober/dns.go#L129 also has metrics for dns.

### Why do you personally want this feature to be implemented?

Maybe we can add these metrics in gatus, and user can scrpe to promethus and make dashboard on Grafana.

### How long have you been using this project?

3 month

### Additional information

If the team agrees to add this feature, I can submit PR to contribute gatus

**Comment by TwiN** [comment]
Sounds reasonable to me, feel free to give it a try!

One thing to keep in mind, however, is that I try to keep Gatus as lightweight as possible, and this extends to metrics too.
See https://status.twin.sh/metrics for an example of what metrics are currently exposed (I believe there's only `gatus_results_total` right now as far as custom metrics goes).

**Comment by wei840222** [comment]
OK, I'll start developing

**Comment by lucagouty** [comment]
@wei840222 thanks for having a look at that one, I'm very interested!

Duration is clearly a must have, but I think we should also add few more : 
- `url`, for now there's only name/key/group
- `cert_valid` / `cert_days_remaining`

I agree with @TwiN, it's great to keep it lightweight, but I'd say those are the bare minimum.

Some inspiration from a project quite similar : https://github.com/louislam/uptime-kuma/blob/master/server/prometheus.js

**Comment by TwiN** [comment]
@lucagouty `cert_valid` and `cert_days_remaining` don't really sound that important to me. At most, you'd just have a graph that gradually goes down. 
Instead, you could just have an endpoint with a condition that uses the `[CERTIFICATE_EXPIRATION]` placeholder, and use the existing `gatus_results_total` metric to figure out when a certificate is about to expire or not.

I do agree on the `url` being a useful label that could be added.

**Comment by lucagouty** [comment]
Hi @TwiN, thank you for taking the time to answer.

> At most, you'd just have a graph that gradually goes down.

Yes I agree, but it's actually quite nice to have it. I know Gatus alerts are very nice, but tbh I intend not to use it and do as much as possible using prometheus/alertmanager/grafana so I keep everything in the same place.

Here is the Grafana dashboard I've made for Uptime Kuma, I'd really love to achieve the same with Gatus :
![image](https://user-images.githubusercontent.com/8852777/164609740-fca8a69a-0395-4296-9a5f-3b00b0eee9a8.png)

I understand you want to keep the metrics as light as possible, but filtering metrics should be quite easy, for example I've done that for Gatus to get rid of process/go metrics I don't need : 
```
metricRelabelings:
- action: drop
  regex: (go|process)_.+
  sourceLabels: [__name__]
```

To give you a bit of context, I've been using Kuma for a while but I had to do a lot of tricks to handle file-based configurations (and [they clearly don't want to ever go that way](https://github.com/louislam/uptime-kuma/issues/270#issuecomment-939416416)) so I'd love to switch to Gatus.

It's up to you though, I don't want to insist, Gatus is already really nice and I've been using it only for a week so I wouldn't want to impose anything on you 😅

**Comment by TwiN** [comment]
@lucagouty I'm not completely against it, but I'd appreciate if you could try to throw a working PoC (doesn't need to be perfect, I just want to see with my own testing configs what the body of the `/metrics` endpoint looks like with these metrics and how much it grows in size) that I can clone and test myself.

Alternatively, you could try and see if @wei840222 is open to the idea of you contributing to his fork, but that's up to you two.

**Comment by lucagouty** [comment]
ok thanks you @TwiN, I think I'll wait for @wei840222 PR to be reviewed, and if that one suits you then I'll try to match what he's done

**Comment by wei840222** [comment]
hi @lucagouty , thx for your interest for this issue.
I still develop in my free time and you can check my pregress here https://github.com/wei840222/gatus.
And I'll try to taking your advice.
In fact our org also build our own uptime dashboard on Grafana and scrape metrics from blackbox exporter, we also want to replace by Gtus

**Comment by wei840222** [comment]
Hi @lucagouty and @TwiN I've prepared the pr, plz review changes
thx for your help :)

**Comment by TwiN** [comment]
Closed by #278
